code/src/java/plugin/exporttokens/SpellMemToken.java

```diff
@@ -153,10 +143,9 @@ public String getToken(String tokenSource, PlayerCharacter aPC, ExportHandler eh
 					{
 						selectedCSpell = charSpellList.get(spellNumber);
 						aSpell = selectedCSpell.getSpell();
-						found = true;
 					}
 				}
-				else if (aObject != null)
+				else
```

Comment:
`aObject` can't be null here: The outer if statement has the conditional `		if ((aObject != null) || (classNum == -1))`

Reply:
null and -1 will pass the || ... right?

Which means it can be null?

Reply:
To be more specific this is the logic:
```
if ((aObject != null) || (classNum == -1)){
			if ((bookName != null) && !bookName.isEmpty())
			{
				if (classNum == -1)
				{
                                 }
----->                      else if (aObject != null)
                       }
}

```

In this case it might be a bit more clear to leave the check, though it will always be true.